### DISCUSSION TITLE
My Contributions

### DISCUSSION THREAD

**Discussion by thelostone-mc** [dcgtc/dgrants#462] [Architecture & Design]
From richard : 

no theres no "my contributions" page / final design yet.  need to know a bit about the specs.  can we show matching rounds for each contribution ? can direct donations and in the same list ?  can i filter between rounds ? sort by highest to lowest ?  can i sort by date ? do we show donation-blocktime, dmY or humanized dates ?  can we do calc from token to USD value ?  will this show a USD value right now or value on donation time ?  can we show donations that failed ? can we show donations that are pending ?   do we go for pagination, infinite scroll, or "load more" button ? can we have a "export all as csv / json" ?  can we search for transaction hashes ?  is all this from a simple graph query with simple orderby, sortby, limit ?   would love to jam with you or philip on this one ! this is crucial as it would effect also other parts of pages later on like a global "/contributions" to see with ALL contribution from everybody .  or  see  "/ethaddress/profile" on a profile page contributions of another user

**Comment by thelostone-mc** [comment]
> can we show matching rounds for each contribution

nope this is not something we'd be able to show 

----
>  can direct donations and in the same list 

I don't think it should.
cc @phutchins#6629+ @bendi#3166 

----
>  can i filter between rounds

Hmm this feels like a useful thing to have but would require additional work. Is there benefit in it ?

----
> sort by highest to lowest ?  can i sort by date

Both

----
> do we show donation-blocktime, dmY or humanized dates ?

Humanized 

----
> can we do calc from token to USD value 

I'd say stick to actual token only cause we don't capture the USD value at the time of contribution which means we'd have to rely on realtime value (aka it would keep changin)

----
> can we show donations that failed 

Hmm I think we should 

----
>  can we show donations that are pending 

I think we should

----
> do we go for pagination, infinite scroll, or "load more" button 

@phutchins / @mk1r / @gdixon do we have a preference here 

>  can we search for transaction hashes

seems like an overkill at this point when they could get this info from blockexplorer. Is there a benefit to this ?
@apbendi @mds1 

> is all this from a simple graph query with simple orderby, sortby, limit ?

I'm assuming we'd have to rely on the graph for the long run cc @gdixon

**Reply by metafraction** [reply]
I don't have a strong preference on pagination, infinite scroll, or "load more" but I feel like 'load more' would be better and load faster initially
